### PR TITLE
Add is constant pattern snippet for null check.

### PR DESCRIPTION
**Pull request by WayneGoosen**
## Summary

Added null check with `is` statement snippet WIP.

Fixes dotnet/docs#6855

### CODE REVIEW

File: snippets/csharp/language-reference/keywords/is/is-const-pattern11.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        object o = null;
+
+        if (o is null)
+            Console.WriteLine("'is' constant pattern 'null' check : True");
+        
+        if (object.ReferenceEquals(o, null))
+            Console.WriteLine("object.ReferenceEqual 'null' check : True");
+
+        if (o == null)
+            Console.WriteLine("Equality operator (==) 'null' check : True");
```
Comment (BillWagner) [inline snippets/csharp/language-reference/keywords/is/is-const-pattern11.cs]: I recommend a second example here, using nullable types. A great way to use this syntax is something like:

```csharp
int? val = someMethodCall();

if (!(val is null))
{
   // Safely use val here.
}
```

Comment (pkulikov) [reply snippets/csharp/language-reference/keywords/is/is-const-pattern11.cs]: @BillWagner with negation, the `is null` pattern looks less readable than using `!=` operator.

Either
```csharp
int? val = SomeMethodCall();
if (val is int v)
{
    // Use v here.
}
```

or
```csharp
int? val = SomeMethodCall();
if (val != null)
{
    // Safely use val.Value here.
}
```

seems to be more readable with nullable types than
```csharp
int? val = SomeMethodCall();
if (!(val is null))
{
    // Safely use val.Value here.
}
```

Comment (BillWagner) [reply snippets/csharp/language-reference/keywords/is/is-const-pattern11.cs]: @pkulikov Yeah, that was a bad example.  maybe something more like:

```csharp
int? customerID  = findCustomerByName(firstName, lastName);
if (customerID  is null)
{
   customerID = registerCustomer(firstName, lastName);
}
```

Comment (pkulikov) [reply snippets/csharp/language-reference/keywords/is/is-const-pattern11.cs]: @BillWagner that's a much better example; thank you.
I would nit pick about details: "Please register" in the raised message and immediate registration afterwards doesn't play well together, but idea is clear.

**Comment by WayneGoosen** [comment]
Once i added the nullable type sample, it became clear i could write a consistent sample between reference and nullable type. Please let me know if you feel the comparison of null checks was valuable and I will bring it back into the sample

File: snippets/csharp/language-reference/keywords/is/is-const-pattern11.cs (hunk 1 above)
Comment (WayneGoosen) [reply snippets/csharp/language-reference/keywords/is/is-const-pattern11.cs]: I have the following as the sample now:

```csharp
object o = null;

if (o is null)
{
    Console.WriteLine("o does not have a value");
}
else
{
    Console.WriteLine($"o is {o}");
}

int? x = 10;

if (x is null)
{
    Console.WriteLine("x does not have a value");
}
else
{
    Console.WriteLine($"x is {x.Value}");
}

// The example displays the following output:
//  o does not have a value
//  x is 10
```
Once i added the nullable type sample, it became clear i could write a consistent sample between reference and nullable type. Please let me know if you feel the comparison of null checks was valuable and I will bring it back into the sample

**Comment by BillWagner** [comment]
@WayneGoosen Thanks for updating this sample.  I like what you added.

> It became clear i could write a consistent sample between reference and nullable type. Please let me know if you feel the comparison of null checks was valuable and I will bring it back into the sample.


That's an important concept for readers to understand. It would be great to add this to the sample.  Once you're ready, remove the "wip" tag from the PR title, and I'll give it one final review and merge.

**Comment by BillWagner** [comment]
Thanks for making those additional updates @WayneGoosen 

This looks great, and I'll :shipit: now.  You should see the changes on the live site on our next regular publishing cycle, no later than Monday morning.

Thanks again for contributing to docs!

**Pull request merged**